src/flamegraph/color/palettes.rs

```diff
@@ -99,6 +99,25 @@ pub(super) mod wakeup {
     }
 }
 
+pub(super) mod rust {
+    use crate::flamegraph::color::BasicPalette;
+
+    // Partition into user code and Rust system code
+    pub fn resolve(name: &str) -> BasicPalette {
+        if name.starts_with("core::") || name.starts_with("std::") {
+            return BasicPalette::Orange;
+        } else if name.contains("as core::")
+            || name.contains("as std::")
+            || name.contains("as alloc::")
+        {
+            return BasicPalette::Orange;
+        } else if name.contains("::") {
+            return BasicPalette::Yellow;
+        }
+        BasicPalette::Red
```

Comment:
I wonder if we want a different color here since the base color is `Orange`, just to improve the contrast. Might be helpful to take a look at an actual rendering of the resulting SVG for comparison.

Reply:
Good point - I'll try something with more contrast.

Reply:
Here's the nicest color combo I've found so far.  I'm starting with the idea that the Rust code has to be orange so it looks "oxidized".  :)

![flamegraph](https://user-images.githubusercontent.com/184890/116882924-fec9b780-abf2-11eb-839c-d071da058fec.png)

Reply:
Right, but which "Rust" code is more clearly Rust, the user's code or the std/core/alloc code? :p
I think what I'd do is make std/core/alloc red, make user code (`::`) orange, and make everything else either green or yellow. I think that might give a nice "all of this is Rust-related, and these bits are not".

Reply:
It's fun playing with these combos.  Here's the latest:
![flamegraph](https://user-images.githubusercontent.com/184890/116997417-c5548300-acaa-11eb-8785-ddd7af2f3521.png)

Reply:
I worry that with the brightness adjustments, red and orange are just too hard to distinguish 😢 Maybe your instinct to use `Yellow` was better in the first place. I'm honestly torn here!

Also, just after suggesting it, I realized that red+green is a bad choice, since it's the most common form of color-blindness. We should probably use basically any color other than green for the non-Rust things. Like... blue. I guess that brings us back to your original proposal 😅 

This is very much bikeshedding at this point. But the fun kind! My overall aims with the colors are:
 - Make user code very visible.
 - Make std code distinguishable from user code.
 - Make user and std colors mesh well, since they're likely to be interspersed (like in your posting).
 - Make non-Rust frames highly distinguishable from Rust frames.
 - Make non-Rust frames more muted than Rust frames, since we're trying to _highlight_ Rust code with this palette.

Working backwards from that, how about:
 - Rust user: Aqua
 - Rust std: Orange
 - Non-Rust: Yellow

Reply:
Heh, I hear you on all that, and I like this latest palette.  The aqua pops right out but matches well with the orange (it's complementary color).  The yellow stands back a bit but also pairs with it's nearby orange to anchor down the non-user lower-level functions (rust std + non-rust).

As an aside, I'm also grappling with a rather short experience with flamegraphs so far, so I'm glad you wanted to explore these choices a bit more.  Also, I'm realizing that I've been favoring certain color palettes because of the font color contrast (the red is a bit hard for me), but I'm drifting off-topic, and I might have another idea about that for another day..

Here's another visual:
![flamegraph-aqua](https://user-images.githubusercontent.com/184890/117078573-1f376600-ad08-11eb-945e-209f2c555fc5.png)

Reply:
This looks great! I'm not a huge fan of the "grenness" of the yellow color, but that's a problem for another day. And so is font contrast :)